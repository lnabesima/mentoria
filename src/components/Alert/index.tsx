import React from 'react';
/** @jsxRuntime classic */
/** @jsx jsx */
import { jsx } from '@emotion/react';
import CircleIcon from '@mui/icons-material/Circle';
import style from './styles';

export enum AlertStatus {
  Ok = 'OK',
  Warning = 'WARNING',
  Critical = 'CRITICAL',
}

type AlertColorMap = Record<string, 'success' | 'primary' | 'error'>

const colorMap: AlertColorMap = {
  [AlertStatus.Ok]: 'success',
  [AlertStatus.Warning]: 'primary',
  [AlertStatus.Critical]: 'error',
};

const alertSizeMap = {
  small: { height: '12px', width: '12px' },
  medium: { height: '24px', width: '24px' },
  large: { height: '48px', width: '48px' },
}

interface IProps {
  status: AlertStatus;
  size?: 'small' | 'medium' | 'large';
}

const Alert: React.FC<IProps> = (props) => {
  const { status, size } = props;

  const styleProps = size ? alertSizeMap[size] : alertSizeMap.small;

  return (
    <CircleIcon css={style(styleProps)} color={colorMap[status]} />
  );
};

Alert.defaultProps = {
  size: 'small',
};
  
export default Alert;
