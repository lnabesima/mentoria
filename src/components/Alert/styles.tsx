/** @jsxRuntime classic */
/** @jsx jsx */
import { css, jsx, keyframes } from '@emotion/react';

export interface AlertSize {
  height: string;
  width: string;
}

const pulse = keyframes`
  0% {
    box-shadow: 0 0 0 0px #CCC;
  }
  100% {
    box-shadow: 0 0 0 3px rgba(0, 0, 0, 0);
  }
`;
 
const style = (props: AlertSize) => css`
  width: ${props.width};
  height: ${props.height};
  border-radius: 50%;
  box-shadow: 0px 0px 1px 1px #CCC;
  animation: ${pulse} 2s infinite;
`;

export default style;
