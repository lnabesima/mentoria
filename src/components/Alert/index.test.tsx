import React from 'react';
import customRender from  '../../test/theme';

import Alert, { AlertStatus } from './index';

describe('[ALERT]',  () => {
  it('should render with default props', () => {
    const wrapper = customRender(<Alert status={AlertStatus.Ok} />)

    expect(wrapper.baseElement).toMatchSnapshot();
  })

  it('should render with custom props', () => {
    const wrapper = customRender(<Alert status={AlertStatus.Warning} size="medium" />)

    expect(wrapper.baseElement).toMatchSnapshot();
  })
});
