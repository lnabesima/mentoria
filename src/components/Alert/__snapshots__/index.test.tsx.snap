// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ALERT] should render with custom props 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-17y33vk-MuiSvgIcon-root-style"
      data-testid="CircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
      />
    </svg>
  </div>
</body>
`;

exports[`[ALERT] should render with default props 1`] = `
<body>
  <div>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-t8w6mh-MuiSvgIcon-root-style"
      data-testid="CircleIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2z"
      />
    </svg>
  </div>
</body>
`;
